### PR TITLE
Fix qwen moe Load balancing loss calculation outside training

### LINKED ISSUE #42100: Qwen3Moe models have bug during generation since they calculate unnecessary load_balancing_loss
### System Info

Qwen3Moe models will calculate load_balancing_loss during evaluation, which will cause bug during generation on the 2nd (and later) steps. 

The problem can be handled by modifying 

```python
if output_router_logits:
    aux_loss = load_balancing_loss_func(
    ...
```

to

```python
if output_router_logits and self.training:
    aux_loss = load_balancing_loss_func(
    ...
```

### Who can help?

_No response_

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Perform model.generate() using a Qwen3MoeForCausalLM model. 

### Expected behavior

Generate successfully.

### PR DESCRIPTION
**Pull request by diegoakel**
# What does this PR do?

Removes the calculation of load balancing loss for Qwen MoE models when outside training mode.

Fixes #42100

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@Rocketknight1

### CODE REVIEW

**Comment by Rocketknight1** [comment]
Posted in #42100 as well, but do you know how `output_router_logits` is getting set? This should only happen when that's set to `True`.
